### PR TITLE
[v2] Additional fixes for tests

### LINKED ISSUE #3324: [v2] fix tests that fail only when run locally
I think this is the remaining tests that needs to be solved, but it is down to 3 min:

Screenshot 2025-10-12 at 17 56 52

[...]

It seems to only be a local error, though (I suspect it's something to do with the numpy version, seed, or something like that). Anyway, that shouldn't happen, so I will fix that. However, assuming this runs fine - do feel free to merge (and then I will do it in another PR)

_Originally posted by @KennethEnevoldsen in https://github.com/embeddings-benchmark/mteb/issues/3306#issuecomment-3394764102_

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
- [x] removed duplicate test testing for MIEB models + encoders the current one is in `test_integrations/test_image_model_task.py`
- [x] figure out where additional  files come from and send them to tmp_path
- [x] fix test which fails locally only by introducing a new model `mteb/random-baseline` that produced a random vector which is consistent on the input
- [x] minor fixes to cases which didn't handle numpy as output
- [x] removed test which tests that private datasets are in fact listed as private (we have the inverse)
- [x] reduced grid size of `test_encoder_dtype_on_task`
- [x] combined tests for metadata so that we have ~1k test instead of ~3k for metadata (runs a bit faster)
- [x] reduced task_grid to only include one classification task and switched out the classification task for a smaller one
- [x] replaced test_cli classification task with a smaller one

All of this combined put us under 3 mins locally (hot cache)

Screenshot 2025-10-12 at 21 09 54

fixes https://github.com/embeddings-benchmark/mteb/issues/2957
fixes #3324 

Added the `mteb/random-baseline` as a fully fledged model, as I could imagine it being relevant as a baseline as well.

### CODE REVIEW

File: mteb/models/model_implementations/random_baseline.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if s is None:
+            s = ""
+        # Convert string to a numeric seed
+        seed = int.from_bytes(s.encode("utf-8"), byteorder="big") % (
+            2**32
+        )  # numpy rng seed must be between 0 and 2**32
+        rng = np.random.default_rng(seed)
+        return rng.random(size)
+
+    def similarity(
```
Comment (Samoed) [inline mteb/models/model_implementations/random_baseline.py:76]: I think this would be automatically cosine based on `model_meta`

File: tests/test_tasks/test_predictions.py [hunk 2]
```diff
@@ -22,62 +21,62 @@
...
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
+                [0, 0],
```
Comment (Samoed) [inline tests/test_tasks/test_predictions.py:45]: This is strange that output of classification task only 0

File: mteb/models/model_implementations/random_baseline.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import numpy as np
+from torch.utils.data import DataLoader
+
+from mteb.abstasks.task_metadata import TaskMetadata
+from mteb.models import Encoder
+from mteb.models.model_meta import ModelMeta
+from mteb.types._encoder_io import Array, BatchedInput, PromptType
+
+
+class RandomBaseline(Encoder):
```
Comment (Samoed) [inline mteb/models/model_implementations/random_baseline.py]: Great addition! We can reuse it in mock models. Also close https://github.com/embeddings-benchmark/mteb/issues/2957

File: mteb/models/model_implementations/random_baseline.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: Hmm it fails if it is not there. Do we add the similarity function somewhere if it is not?

Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: (might be because I am inheriting from Encoder)

File: tests/test_tasks/test_predictions.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply tests/test_tasks/test_predictions.py:45]: Hmm yeah can't tell you why though

File: mteb/models/model_implementations/random_baseline.py (hunk 1 above)
Comment (Samoed) [reply mteb/models/model_implementations/random_baseline.py:76]: Yes, I've added in absencoder https://github.com/embeddings-benchmark/mteb/blob/60adf65d542d2de075f759ed32fd1d4f38618b96/mteb/models/abs_encoder.py#L237

File: tests/test_integrations/test_dtype_encoders.py [hunk 4]
```diff
@@ -7,20 +7,54 @@
...
+        MockRerankingTask(),
+        MockRetrievalTask(),
+        MockSTSTask(),
+        MockMultilabelClassification(),
+        MockSummarizationTask(),
+        MockInstructionRetrieval(),
+        MockInstructionReranking(),
+        MockRetrievalDialogTask(),
+        MockTextZeroShotClassificationTask(),
+    ],
```
Comment (Samoed) [inline tests/test_integrations/test_dtype_encoders.py]: Why not to use mock task test grid here?

File: mteb/models/model_implementations/random_baseline.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        *,
+        task_metadata: TaskMetadata,
+        hf_split: str,
+        hf_subset: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> Array:
+        all_embeddings = []
+        for batch in inputs:
+            for text in batch["text"]:
```
Comment (Samoed) [inline mteb/models/model_implementations/random_baseline.py]: Maybe create for each input embedding to make random model multimodal?

File: tests/test_integrations/test_dtype_encoders.py (hunk 4 above)
Comment (KennethEnevoldsen) [reply tests/test_integrations/test_dtype_encoders.py]: This is mostly to test the flow of dtypes throughout the codebase. Multilingual subsets generally use the same underlying flow (evaluate_subset) so we don't need to test it twice. This reduced the size of the grid quite a lot, given that we are testing multiple models

Comment (Samoed) [reply tests/test_integrations/test_dtype_encoders.py]: Maybe we can create grid with these tasks and then combine them with multilingual to full grid?

File: mteb/models/model_implementations/random_baseline.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: Right, so it only works when you inherit from AbsEncoder? I would prefer keeping this function independent. It has already revealed some issues with similarity being assumed to be a torch tensor

File: mteb/models/model_implementations/random_baseline.py (hunk 5 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py]: I am all for doing this in another PR when relevant

File: mteb/models/model_implementations/random_baseline.py (hunk 1 above)
Comment (Samoed) [reply mteb/models/model_implementations/random_baseline.py:76]: > Right, so it only works when you inherit from AbsEncoder?

Yes

> Right, so it only works when you inherit from AbsEncoder?

Leave only in `AbsEncoder`? Don’t we support both numpy and torch?

File: tests/test_integrations/test_dtype_encoders.py (hunk 4 above)
Comment (KennethEnevoldsen) [reply tests/test_integrations/test_dtype_encoders.py]: agree - fixed!

File: mteb/models/model_implementations/random_baseline.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: In this PR, I fixed a few cases that didn't work if the output of similarity was a numpy vector. 

Maybe the MockEncoders shouldn't inherit from AbsMockEncoder?

Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: I also see that we have `MockSentenceTransformersbf16Encoder` which I would probably convert to `MockSentenceTransformersbf16Encoder` > `MockTorchbf16Encoder`

Comment (Samoed) [reply mteb/models/model_implementations/random_baseline.py:76]: Maybe, I want to refactor test models a bit

Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py:76]: alright I will leave it for another PR

File: mteb/models/model_implementations/random_baseline.py (hunk 5 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/random_baseline.py]: Added this here as it was needed for there merge
